docs/adrs/006-web-apis.md

```diff
@@ -0,0 +1,171 @@
+
+Creating the `client` should be the starting point for any application working with Penumbra:
+
+```ts
+import { createPenumbraClient } from '@penumbra-zone/client';
+
+export const client = createPenumbraClient();
+```
+
+The flow of work with the `client` would be as follows:
+
+- User creates an instance of the `client`
+- They use the `client` to establish the injected connection, and the `client` saves the connection in its state
+- To fetch information from the wallet, the application passes the `client` instance to the fetch methods
+
+The idea of a single shared `client` is inspired by these popular TypeScript libraries:
+
+- [Prisma ORM](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/instantiate-prisma-client)
+- [Supabase](https://supabase.com/docs/reference/javascript/initializing)
+- [Viem](https://viem.sh/docs/clients/intro)
+
+## Injected connection
+
+The `client` must be able to connect to injected wallets. To make the API possible, the interface injected into the `window` object must be standardized:
+
+```ts
+export interface PenumbraProvider extends Readonly<PenumbraStateEventTarget> {
+  /** Should contain a URI at the provider's origin,
+   * serving a manifest describing this provider */
+  readonly manifest: string;
+
+  /** Call to acquire a `MessagePort` to this provider, asks user for approval if wasn't connected before */
+  readonly connect: () => Promise<MessagePort>;
+
+  /** Call to indicate the provider should discard approval of this origin. */
+  readonly disconnect: () => Promise<void>;
+
+  /** Should synchronously return the present connection state.
+   * - `true` indicates active connection.
+   * - `false` indicates connection is inactive.
+   */
+  readonly isConnected: () => boolean;
+
+  /** Synchronously return present injection state */
+  readonly state: () => PenumbraState;
+
+  /** Standard EventTarget methods emitting PenumbraStateEvent upon state changes */
+  readonly addEventListener: PenumbraEventTarget['addEventListener'];
+  /** Unsubscribes from the state change events. Provide the same callback as in `addEventListener` */
+  readonly removeEventListener: PenumbraEventTarget['removeEvenListener'];
+}
+```
+
+## Client API
+
+A globally shared `client` instance should be able to establish the connection with
+injected wallet providers and use the connection to query the blockchain. The API in this case would be:
+
+```ts
+import type { PenumbraService } from '@penumbra-zone/protobuf';
+import type { PromiseClient } from '@connectrpc/connect';
+
+interface PenumbraClient {
+  /**
+   * Connects to the injected provider, asks for user approval if wasn't connected before.
+   * If `providerUrl` argument is not provided, tries to connect to the first injected provider.
+   * Returns the manifest URL of the connected provider or throws an error otherwise.
+   */
+  readonly connect: (providerUrl?: string) => Promise<string>;
```

Comment:
returning manifest url from the `connect` method seems like the least useful choice.

it seems more better to return the `MessagePort` or a created `Transport`, allowing the caller to handle the `Transport` directly if they wish, as designed in #1648

if the manifest is really desired, just go ahead and provide a parsed manifest - the connection init process must fetch it anyway to confirm it exists.

Reply:
That's true – the provider origin URL is not that useful. The same can be said about `MessagePort` or `Transport` – they are of no value for the library users. It might be better to return `Promise<void>` for now until we find out the  really useful return, to not produce breaking changes in the future

Reply:
changed to return `Promise<void>`